### PR TITLE
check for `go generate` on CircleCI

### PR DESCRIPTION
**Pull request by tbg**
this guarantees that no commit for which `go generate ./...` modifies or
deletes any files passes the test.

### CODE REVIEW

**Comment by petermattis** [comment]
LGTM

File: circle.yml [hunk 1]
```diff
@@ -30,6 +30,8 @@ dependencies:
 
 test:
   override:
+    # Check whether the committer forgot to run `go generate`.
+    - docker run cockroachdb/cockroach-dev shell "go generate ./... && git ls-files --modified --deleted | diff /dev/null -"
```
Comment (bdarnell) [inline circle.yml]: I think the git ls-files call should also include --others to find new/untracked files.

**Comment by bdarnell** [comment]
LGTM (I also see that the github change to treat .pb.go files as generated has gone live)

File: circle.yml (hunk 1 above)
Comment (tschottdorf) [inline circle.yml]: That'll require some more changes since we add cockroach.sh in the directory, but I'll just move that to the root before I merge this.

Comment (bdarnell) [inline circle.yml]: Or you can add cockroach.sh to .gitignore (with a comment about when this occurs).

Comment (tschottdorf) [inline circle.yml]: done.

**Pull request merged**